tests/metrics/test_metrics.py

```diff
@@ -190,7 +190,9 @@ def test_saving_pickable(tmpdir, metric: Metric):
     assert results_before_save == results_after_load
 
 
-def check_call_order():
+def test_correct_call_order():
+    """ Check that hooks are called in the expected order """
+
     class DummyMetric(Metric):
```

Comment:
this would be too much to change heere, but I just want to comment that I think the right way to go about checking call order etc. is with the python unittest mocking system. it does not require us to write these artificial classes that track calls, because it's built in. something to look out for in the future I think. Also in other places of Lightning. cc @justusschock @Borda

Reply:
Lets do that in a follow up PR (not that I have that much experience with mocking)